diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -74,7 +241,202 @@ class LineLength : Rule("line-length") {
+                    }
+                    .forEach {
+                        searchBinaryExpression(it, binList)
+                    }
+        } else {
+            binList.add(node)
+            binList.add(node.treeParent.findChildByType(PREFIX_EXPRESSION) ?: return)
+        }
+    }
+
+    private fun dfsForProperty(node: ASTNode, binList: MutableList<ASTNode>) {
+        node.getChildren(null).forEach {
+            if (it.elementType in PROPERTY_LIST)
+                if (it.elementType == REFERENCE_EXPRESSION && it.treeParent.elementType == CALL_EXPRESSION)
+                    binList.add(it.treeParent)
+                else
+                    binList.add(it)
+            dfsForProperty(it, binList)
+        }
+    }
+
+    private fun fixLongString(fixableType: LongLineFixableEnum) {
+        val wrongNode = fixableType.node
+        val leftOffset = fixableType.leftOffset
+        val lineLength = fixableType.configuration.lineLength
+        val binList = fixableType.binList
+        val allText = binList.joinToString("") { it.text }
+        var binaryText = ""
+        binList.forEachIndexed { index, astNode ->
+            binaryText += astNode.text
+            if (STRING_PART_OFFSET + leftOffset + binaryText.length > lineLength) {
+                if (astNode.elementType == LITERAL_STRING_TEMPLATE_ENTRY) {
+                    val lastCharIndex = lineLength.toInt() - leftOffset - STRING_PART_OFFSET
+                    val indexLastSpace = (allText.substring(0, lastCharIndex).lastIndexOf(' '))
+                    if (indexLastSpace == -1)
+                        return
+                    splitTextAndCreateNode(wrongNode, allText, indexLastSpace)
+                    return
+                } else if (index == 0) {
+                    return
+                }
+                splitTextAndCreateNode(wrongNode, allText, binaryText.length)
+                return
+            }
+        }
+    }
+
+    private fun createNodes(fixableType: LongLineFixableEnum) {
+        val node = fixableType.node
+        val indexLastSpace = fixableType.indexLastSpace
+        val text = fixableType.text
+        splitTextAndCreateNode(node, text, indexLastSpace)
+    }
+
+    private fun splitTextAndCreateNode(node: ASTNode, text: String, index: Int) {
+        val resultText = "\"" + text.substring(0, index) + "\" +\n\"" + text.substring(index) + "\""
+        val newNode = KotlinParser().createNode(resultText)
+        node.removeChild(node.findChildByType(STRING_TEMPLATE)!!)
+        val prevExp = CompositeElement(BINARY_EXPRESSION)
+        node.addChild(prevExp, null)
+        prevExp.addChild(newNode, null)
+    }
+
     class LineLengthConfiguration(config: Map<String, String>) : RuleConfiguration(config) {
         val lineLength = config["lineLength"]?.toLongOrNull() ?: MAX_LENGTH
     }
+
+    enum class LongLineFixableEnum {
+        COMMENT,
+        CONDITION,
+        ERROR,
```

Comment:
```suggestion
        NONE,
```
I mean, it's not an error, if we can't fix it.